src/main/rx/xbus.c

```diff
@@ -224,8 +227,14 @@ static void xBusDataReceive(uint16_t c)
     }
 
     // Check if we shall start a frame?
-    if ((xBusFramePosition == 0) && (c == XBUS_START_OF_FRAME_BYTE)) {
-        xBusDataIncoming = true;
+ if (xBusFramePosition == 0)	{
+   	if (c == XBUS_START_OF_FRAME_BYTE_A1) {
+   		xBusDataIncoming = true;
+   		xBusFrameLength = XBUS_FRAME_SIZE_A1;	//decrease framesize (when receiver change, otherwise board must reboot)
+   	}   	else if (c == XBUS_START_OF_FRAME_BYTE_A2) {//16channel packet
+    		xBusDataIncoming = true;
+    		xBusFrameLength = XBUS_FRAME_SIZE_A2;	//increase framesize
+   	}
     }
```

Comment:
I almost think xbusDataincoming bool is superfluous - if the frame length is set then it's data incoming no? Just need to remove the initial setting to A1 from init and set to 0. Is it dual purposing it though... @martinbudden your thoughts?